Source updates, usage tagging

stepps00 opened the pull request:
Fixes https://github.com/whosonfirst/whosonfirst-sources/issues/97, #40, #113, and #114 .

Also includes a few minor tweaks to the README, `docs.py` and source json files. Includes:

- Added `mz:is_current` and `edtf:deprecated` property descriptions to the README. Similar descriptions to what is found in the [properties repo](https://github.com/whosonfirst/whosonfirst-properties/) repo. 

- Updated the `docs.py` script to include logic to write out "This source has been deprecated." if an `edtf:deprecated` property has been set.

- Added a `mz:is_current` and `edtf:deprecated` property to `addr`, `geom`, `lbl` and `name` json files, as these are not sources (they're catalogued in the properties repo)

- Added description to `lbl` json, for consistency

- Crawled the repo and appended values to the `usage:*` properties for each json file in the repo. Running the makefile added the usage tags to each source in the sources/README file
  - There may be additional work to do here. If a usage tag was missed, prefer merging and submitting a follow-up PR.

- Adds `src:via` property to three json sources. These are lists of sources, the key being the source name, value being the source website (not all are included yet). 

- Updated the `docs.py` to write out a list of each "via" source in the README (see Quattroshapes). 

- Updated template file to include new potential properties.

Note that most of the json files have large diffs. This is due to all of the property keys being sorted when writing back to disc. The properties are now alphabetized - I can wire this into the makefile if helpful.

nvkelso commented:
Also, for these that were deprecated (`addr`, `geom`, `lbl` and `name`), indicate in their JSON description that they are properties tracked in the properties repo instead – provide a link to over there.

nvkelso commented:
I'm not seeing usage for YS, please make sure you're looking for any `src:*` instead of a list of specific sources like `src:geom`. This should be a usage for value of a property, in this case label centroids.

nvkelso commented:
Should we be tracking what placetypes are covered by this source?

Other sources without usage:

- `acgov`
- `acme`
- `arg-caba`
- `atgov`
- `bebrus`
- `btv` - was this Vermont neighbourhood polygons?
- `can-databc` - indicate this is used for clip shorelines for Vancouver neighbourhood geoms?
- `can-vicodc`
- `chgov`
- `dial` (expecting concordance)
- `dk-geodk`
- `ds` (expecting concordance)
- `ebc` (is this British Columbia, Canada constituencies?!)
- `edtf` - we should indicate this is a WOF-ism to indicate "when" certain actions were taken.
- `gec` (fips)
- `geom` – shouldn't this say property?
- `hsgov`
- `ie-gov`
- `iso` – why is this also `property` instead of just `concordance`?
- `itu` – why is this also `property` instead of just `concordance`?
- `lacity_oof
- `lieu`
- `localwiki`
- `mz` - has an extra space in the description causing it to not italicize BEGINNING. 
- `meso`
    - This source includes `CC-BY compatible` data from the following organizations:
       _Please add the `CC-BY compatible` text as indicated above._
- `mudf`  - has an extra space in the description causing it to not italicize END. 
    - no usage at all?
- `no-geonorge`
- `nycgov`
- `nycgov_dca`
- `nycgov_dohmh`
- `nycgov_subway`
- `os`
- `osm` – this one is important to say usage concordances!
- `out` – I assume because it's only used in venues repos?
- `qs` - look at all those sources of sources, yeah!
    - We monkeyed with this in person to set context, source name, and optional link, desc
- `se-lant`
- `sfac`
- `sfgov_rbl`
- `smcgov`
- `statcan`
- `svn-sma`
- `thelist` - this is venues only?
- `transitland` - this is venues only?
- `uscensus`
- `uszcta` - probably postalcodes?
- `ys`

stepps00 commented on bin/docs.py:
```diff
@@ -52,6 +52,10 @@
 
+        if details.get('edtf:deprecated'):
+            if not details['edtf:deprecated'] == 'uuuu':
+                docs.write("* %s\n" % ('This source has been deprecated.'))
```
https://github.com/whosonfirst/whosonfirst-sources/pull/110/commits/f57f9fd637d06d18d454ceb8638a290c30b27043

stepps00 commented on README.md:
```diff
@@ -40,6 +40,10 @@ While a source .json file in the `whosonfirst-sources` repository does not requi
 
+* `"mz:is_current":` Represents whether of not a source is currently in use (_string, optional property_). `0` signifies "not current".
+
+* `"edtf:deprecated":` Indicates the date when a place was determined to be invalid, was never a "going concern" (_string, optional property_). _Format: YYYY-MM-DD (though these dates can be encoded with any valid EDTF syntax)_.
```
https://github.com/whosonfirst/whosonfirst-sources/pull/110/commits/087f17bb094dc6c95916c650aa84dc0d37423369

stepps00 commented:
https://github.com/whosonfirst/whosonfirst-sources/pull/110#issuecomment-390714135 fixed in https://github.com/whosonfirst/whosonfirst-sources/pull/110/commits/4a3c3e334b055c79fc3945607e05f69b01757699

nvkelso commented:
`src:via` has 3 more groups to de-cluster?

- Chile: Global Map of Chile: International Steering Committee for Global Mapping / Instituto Geografico Militar de Chile
- Palestine: Global Map of Palestine: ISCGM/Palestinian National Authority
- South Africa: Global Map of the Republic of South Africa: ISCGM/State Copyright

stepps00 commented:
https://github.com/whosonfirst/whosonfirst-sources/pull/110#issuecomment-390717528 taken care of in https://github.com/whosonfirst/whosonfirst-sources/pull/110/commits/9c3ff4fa7a1a10a9cf98c37c35c4e82d7601ad45

stepps00 commented:
https://github.com/whosonfirst/whosonfirst-sources/pull/110/commits/9c3ff4fa7a1a10a9cf98c37c35c4e82d7601ad45 taken care of in https://github.com/whosonfirst/whosonfirst-sources/pull/110#issuecomment-390736614

Notes:

There are several sources not in the `whosonfirst-data` repo that are catalogued in the `whosonfirst-sources` repo. This is because the sources are used on postalcodes, venues, or are in a PR/upcoming PR. Notes about the sources you listed above:

Updated usage keys for:
- `acgov`
- `arg-caba`
- `atgov`
- `can-databc`
- `chgov`
- `ebc`
- `gec`
- `itu`
- `lacity_oof`
- `nycgov_dca`
- `nycgov_dohmh`
- `osm`
- `out`
- `qs`
- `se-lant`
- `sfac`
- `sfgov_rbl`
- `smcgov`
- `statcan`
- `thelist`
- `transitland`
- `uscensus`
- `uszcta`
- `ys`

Added usage key, but the source is not yet in use (in PR):
- `btv`
- `can-vicodc`
- `dk-geodk`
- `ie-gov`
- `svn-sma`
- `no-geonorge`

Unable to find concordance, property, or geometry usage for:
- `acme`
- `can-vicodc`
- `hsgov`
- `lieu`
- `nycgov_subway`
- `os`
- `localwiki`
- `nycgov`
- `bebrus`

Deprecated:
- `dial`: should not have been added.. this is the same as `itu`
- `ds`: in statoids prop, but not in a concordance/prop/geom
- `edtf`: added link to prop repo
- `geom`: added link to prop repo

No change, correct as is:
- `iso`: we store `iso:country` property and iso concordance.

Updated minor spacing issue:
- `mz`
- `mudf`

Other:
- `meso`: Updated the `docs.py` to include that text

stepps00 commented on source_template.json:
```diff
@@ -8,8 +8,11 @@
 	"license": "",
 	"license_type": "",
 	"license_text": "",
+    "src:via": {"Whos On First":"https://whosonfirst.org/"},
```
6119686

stepps00 commented:
https://github.com/whosonfirst/whosonfirst-sources/pull/110#issuecomment-390751331 taken care of in 
22c6a2b

stepps00 commented:
@nvkelso - all comments should be addressed in this PR. A few notes:

- We have deprecated the `name`, `addr`, `geom`, and `lbl` property prefixes and pointed to their json files in the properties repo. There are three other properties catalogued, though.. `osm`. `edtf`, and `lieu`. I suggest keeping `osm` because it is used as a concordance value in some records, but deprecating `lieu` and `edtf` and pointing to their json files in the properties repo. Thoughts?

- Follow-ups are being tracked in https://github.com/whosonfirst/whosonfirst-sources/issues/98

nvkelso commented:
I’m fine with that.



> On May 21, 2018, at 16:48, Stephen Epps <notifications@github.com> wrote:
> 
> @nvkelso - all comments should be addressed in this PR. A few notes:
> 
> We have deprecated the name, addr, geom, and lbl property prefixes and pointed to their json files in the properties repo. There are three other properties catalogued, though.. osm. edtf, and lieu. I suggest keeping osm because it is used as a concordance value in some records, but deprecating lieu and edtf and pointing to their json files in the properties repo. Thoughts?
> 
> Follow-ups are being tracked in #98
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

nvkelso commented on bin/docs.py:
```diff
@@ -70,13 +81,24 @@
+            docs.write("\n  This source includes `CC-BY compatible` data from the following organizations:\n")
+            for via in details['src:via']:
+                if via['source_note']:
+                    docs.write("  \t* **%s**: [%s](%s) - %s\n" % (via["context"],via["source_name"],via["source_link"],via["source_note"]))
```
Can you add a test for a link and only add the markdown link decorations if there is a link, please?

Here and one down.

nvkelso commented:
For `Yerbashapes` can you add `src:via` similar to Zetashapes:

- Flickr
- Mapzen
